### PR TITLE
Add validators to sample id

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR adds two validators to the sample_id on the sample sheet model.

- Removes trailing index.
- Enforces LIMS ID format (3 letters + ≥3 digits).

This means that any non valid sample id will cause a validation error when the sample sheet is parsed. As far as I know, there is no reason to allow invalid sample ids in sample sheets. Maybe if it is a really old sample sheet?

Having the validation directly on the `sample_id` property on the sample sheet domain model is useful because it provides guarantees for any access of the property in the code base that the sample id has a valid value.

### Added
- Validation of sample id on sample sheet samples

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by ChrOertlin — APPROVED** [review]
Awesome! This is much better!

File: cg/apps/demultiplex/sample_sheet/models.py [hunk 1]
```diff
@@ -14,12 +17,18 @@
 
 LOG = logging.getLogger(__name__)
 
+SampleId = Annotated[
```
Comment (henrikstranneheim) [inline cg/apps/demultiplex/sample_sheet/models.py:20]: Should this be in a `cg/apps/demultiplex/sample_sheet/validators.py` and imported here? It feels a bit out of place defined directly in this module

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

**Comment by seallard** [comment]
Deployed to stage. Could not setup suitable test data.

**Pull request merged**